openwisp_monitoring/db/backends/influxdb/client.py

```diff
@@ -130,11 +130,15 @@ def read(self, key, fields, tags, **kwargs):
         since = kwargs.get('since')
         order = kwargs.get('order')
         limit = kwargs.get('limit')
+        retention_policy = kwargs.get('retention_policy')
         if extra_fields and extra_fields != '*':
             fields = ', '.join([fields] + extra_fields)
         elif extra_fields == '*':
             fields = '*'
-        q = f'SELECT {fields} FROM {key}'
+        if not retention_policy:
+            q = f'SELECT {fields} FROM {key}'
+        else:
+            q = f'SELECT {fields} FROM {retention_policy}.{key}'
```

Comment:
do not repeat the whole query, follow the DRY principle please:

```python
from_clause = key
if retention_policy:
    from_clause = f'{retention_policy}.{key}'
q = f'SELECT {fields} FROM {from_clause}'
```

or a more compact version:

```python
from_clause = f'{rt}.{key}' if rt else key
q = f'SELECT {fields} FROM {from_clause}'
```